lib/vdsm/storage/fileVolume.py

```diff
@@ -568,7 +568,7 @@ def _allocate_volume(cls, vol_path, size, preallocate):
             # If the image is preallocated, allocate the rest of the image
             # using fallocate helper. qemu-img create always writes zeroes to
             # the first block so we should skip it during preallocation.
-            if preallocate == sc.PREALLOCATED_VOL:
+            if size > 4096 and preallocate == sc.PREALLOCATED_VOL:
```

Comment:
IMHO using `BLOCK_SIZE_4K` instead of 4096 is better (3 times).

Reply:
Not in this case, it just make the code more verbose and it can cause a real bug
since this code depends on qemu-img behavior to allocated 4096 bytes when creating
a new image.

Reply:
No problem.
From the comment `writes zeroes to the first block` I assumed this is always a 4096 bytes' block.
Maybe a more correct solution (even if not in this patch, if you don't want refactoring here) could be to define other constant like `QEMU_IMG_ZERO_BLOCK_SIZE` or something, and to use it.
Anyway, you already have my approval :)

Reply:
Constant like QEMU_IMG_ZERO_BLOCK_SIZE or QEMU_IMG_MIN_ALLOC is the right way to handle this. Feel free to post a PR cleaning this up.